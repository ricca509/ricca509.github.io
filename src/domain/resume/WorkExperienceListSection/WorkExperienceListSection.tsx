import React from "react";
import { WorkExperience } from "./WorkExperience/WorkExperience";
import { SectionTitle } from "@Components/SectionTitle/SectionTitle";
import { ComponentProps } from "@Typings/component";
import { ProfileQuery } from "../../../../graphql-types";

interface WorkExperienceListSectionProps extends ComponentProps {
  experienceList: ProfileQuery["experience"]["edges"];
}

export const WorkExperienceListSection: React.FC<
  WorkExperienceListSectionProps
> = ({ experienceList }) => {
  return (
    <>
      <SectionTitle>Experience</SectionTitle>
      {experienceList.map(({ node }) => {
        return (
          <WorkExperience
            key={node.id}
            company={node?.frontmatter?.company}
            role={node?.frontmatter?.role}
            fromDate={node?.frontmatter?.from_date}
            toDate={node?.frontmatter?.to_date}
            technologies={node?.frontmatter?.technologies}
          >
            {node.html}
          </WorkExperience>
        );
      })}
    </>
  );
};

export default WorkExperienceListSection;
