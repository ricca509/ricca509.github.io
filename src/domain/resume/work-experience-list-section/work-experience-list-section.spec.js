import React from "react";
import { shallow } from "enzyme";
import WorkExperienceListSection from "./";

describe("The work-experience-list-section component", () => {
  it("renders a list of work experiences and a title", () => {
    const props = {
      experienceList: [
        {
          node: {
            id: "aaabbb123",
            frontmatter: {
              company: "Company A",
              location: "London, UK",
              role: "Senior Fullstack Developer",
              from_date: "Mar 2020",
              to_date: "Sep 2020",
              technologies: ["Node.js", "React"],
            },
            html: "<p>Some html here</p>",
          },
        },
        {
          node: {
            id: "aaabbb1234",
            frontmatter: {
              company: "Company B",
              location: "London, UK",
              role: "Senior Fullstack Developer",
              from_date: "Mar 2019",
              to_date: "Sep 2021",
              technologies: ["Node.js", "React"],
            },
            html: "<p>Some other html here</p>",
          },
        },
      ],
    };
    const wrapper = shallow(<WorkExperienceListSection {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
