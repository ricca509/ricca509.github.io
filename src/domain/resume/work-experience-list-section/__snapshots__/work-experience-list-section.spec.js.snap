// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The work-experience-list-section component renders a list of work experiences and a title 1`] = `
<Fragment>
  <SectionTitle>
    Experience
  </SectionTitle>
  <WorkExperience
    company="Company A"
    fromDate="Mar 2020"
    key="aaabbb123"
    role="Senior Fullstack Developer"
    technologies={
      Array [
        "Node.js",
        "React",
      ]
    }
    toDate="Sep 2020"
  >
    &lt;p&gt;Some html here&lt;/p&gt;
  </WorkExperience>
  <WorkExperience
    company="Company B"
    fromDate="Mar 2019"
    key="aaabbb1234"
    role="Senior Fullstack Developer"
    technologies={
      Array [
        "Node.js",
        "React",
      ]
    }
    toDate="Sep 2021"
  >
    &lt;p&gt;Some other html here&lt;/p&gt;
  </WorkExperience>
</Fragment>
`;
